### PR TITLE
plasma-infra: Use Node.js LTS 

### PR DESCRIPTION
**Pull request by Yakutoc**
Плановый переход на Node.js/NPM LTS. 

На момент создания ПР: 

- 18.16.1 (Node)
- 9.5.1 (NPM)

### Проблемы с переходом на Node.js 17 и выше  

[error-0308010c-digital-envelope-routines-unsupported](https://sebhastian.com/error-0308010c-digital-envelope-routines-unsupported/)  

> Command line tools like Webpack uses the “MD4 algorithm” to create file hashes. These file hashes are then used to keep track of changes in your JavaScript files.
> By default, OpenSSL version 3 didn’t enable the MD4 algorithm support. If you upgraded Node.js to version 17 and above, you’ll see this error when building an application using Webpack.

TL;DR

1. To fix this error, you need to upgrade to Webpack 5
2. Add --openssl-legacy-provider flag to your build script
3. If you use Create React App, upgrade react-scripts to v5
4. Downgrade to Node v16 

4 пункт нам не подходит. Скорее всего будем делать 1 и 3.    

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.218.1-canary.594.5472058872.0
  npm install @salutejs/plasma-core@1.123.1-canary.594.5472058872.0
  npm install @salutejs/plasma-hope@1.218.1-canary.594.5472058872.0
  npm install @salutejs/plasma-icons@1.152.1-canary.594.5472058872.0
  npm install @salutejs/plasma-temple@1.167.1-canary.594.5472058872.0
  npm install @salutejs/plasma-ui@1.199.1-canary.594.5472058872.0
  npm install @salutejs/plasma-web@1.218.1-canary.594.5472058872.0
  npm install @salutejs/plasma-sb-utils@0.121.1-canary.594.5472058872.0
  # or 
  yarn add @salutejs/plasma-b2c@1.218.1-canary.594.5472058872.0
  yarn add @salutejs/plasma-core@1.123.1-canary.594.5472058872.0
  yarn add @salutejs/plasma-hope@1.218.1-canary.594.5472058872.0
  yarn add @salutejs/plasma-icons@1.152.1-canary.594.5472058872.0
  yarn add @salutejs/plasma-temple@1.167.1-canary.594.5472058872.0
  yarn add @salutejs/plasma-ui@1.199.1-canary.594.5472058872.0
  yarn add @salutejs/plasma-web@1.218.1-canary.594.5472058872.0
  yarn add @salutejs/plasma-sb-utils@0.121.1-canary.594.5472058872.0
  ```
</details>

:baby_chick: Download canary assets:

[color_asdk_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_ios-swift--canary.594.5472058872.swift)  
[color_asdk_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_kotlin--canary.594.5472058872.kt)  
[color_asdk_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_react-native--canary.594.5472058872.ts)  
[color_asdk_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_xml--canary.594.5472058872.xml)  
[color_plasma_b2c_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_ios-swift--canary.594.5472058872.swift)  
[color_plasma_b2c_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_kotlin--canary.594.5472058872.kt)  
[color_plasma_b2c_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_react-native--canary.594.5472058872.ts)  
[color_plasma_b2c_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_xml--canary.594.5472058872.xml)  
[color_sberHealth_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.594.5472058872.swift)  
[color_sberHealth_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.594.5472058872.kt)  
[color_sberHealth_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.594.5472058872.ts)  
[color_sberHealth_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_xml--canary.594.5472058872.xml)  
[color_sbermarket_business_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.594.5472058872.swift)  
[color_sbermarket_business_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.594.5472058872.kt)  
[color_sbermarket_business_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.594.5472058872.ts)  
[color_sbermarket_business_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_xml--canary.594.5472058872.xml)  
[color_sbermarket_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.594.5472058872.swift)  
[color_sbermarket_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.594.5472058872.kt)  
[color_sbermarket_metro_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.594.5472058872.swift)  
[color_sbermarket_metro_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.594.5472058872.kt)  
[color_sbermarket_metro_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.594.5472058872.ts)  
[color_sbermarket_metro_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_xml--canary.594.5472058872.xml)  
[color_sbermarket_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.594.5472058872.ts)  
[color_sbermarket_selgros_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.594.5472058872.swift)  
[color_sbermarket_selgros_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.594.5472058872.kt)  
[color_sbermarket_selgros_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.594.5472058872.ts)  
[color_sbermarket_selgros_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_xml--canary.594.5472058872.xml)  
[color_sbermarket_wlbusiness_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.594.5472058872.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.594.5472058872.kt)  
[color_sbermarket_wlbusiness_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.594.5472058872.ts)  
[color_sbermarket_wlbusiness_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_xml--canary.594.5472058872.xml)  
[color_sbermarket_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_xml--canary.594.5472058872.xml)  
[color_sberonline_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.594.5472058872.swift)  
[color_sberonline_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.594.5472058872.kt)  
[color_sberonline_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.594.5472058872.ts)  
[color_sberonline_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_xml--canary.594.5472058872.xml)  
[color_sberprime_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.594.5472058872.swift)  
[color_sberprime_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.594.5472058872.kt)  
[color_sberprime_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.594.5472058872.ts)  
[color_sberprime_xml--canary.594.5472058872.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_xml--canary.594.5472058872.xml)  
[shadow_sbermarket_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.594.5472058872.ts)  
[typo_mage_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.594.5472058872.swift)  
[typo_mage_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.594.5472058872.kt)  
[typo_mage_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.594.5472058872.ts)  
[typo_plasma_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.594.5472058872.swift)  
[typo_plasma_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.594.5472058872.kt)  
[typo_plasma_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.594.5472058872.ts)  
[typo_ruler_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.594.5472058872.swift)  
[typo_ruler_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.594.5472058872.kt)  
[typo_ruler_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.594.5472058872.ts)  
[typo_sage_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.594.5472058872.swift)  
[typo_sage_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.594.5472058872.kt)  
[typo_sage_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.594.5472058872.ts)  
[typo_sbermarket_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.594.5472058872.swift)  
[typo_sbermarket_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.594.5472058872.kt)  
[typo_sbermarket_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.594.5472058872.ts)  
[typo_soulmate_ios-swift--canary.594.5472058872.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.594.5472058872.swift)  
[typo_soulmate_kotlin--canary.594.5472058872.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.594.5472058872.kt)  
[typo_soulmate_react-native--canary.594.5472058872.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.594.5472058872.ts)

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: package.json [hunk 1]
```diff
@@ -152,6 +152,6 @@
     "babel-plugin-annotate-pure-calls": "0.4.0"
   },
   "engines": {
-    "npm": "^8.19.0 || 6.14.17"
```
Comment (Yeti-or) [inline package.json:155]: а на 8.19 тоже будет работать?

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-16.19.1
```
[the remark this answers is not in the document]
Comment (Yakutoc) [reply .nvmrc:1]: @Yeti-or 

Хорошо, давай зафиксируем версию.

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-594/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-594/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-594/web-storybook/
ui storybook: http://plasma.sberdevices.ru/pr/pr-594/ui-storybook/
temple storybook: http://plasma.sberdevices.ru/pr/pr-594/temple-storybook/

File: package.json (hunk 1 above)
Comment (Yakutoc) [reply package.json:155]: @Yeti-or Да, проверял связку LTS + npm 8.19.0

Сносил все, и ставил заново

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-594/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-594/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-594/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-594/web-storybook/
ui storybook: http://plasma.sberdevices.ru/pr/pr-594/ui-storybook/
temple storybook: http://plasma.sberdevices.ru/pr/pr-594/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by kayman233 — APPROVED** [review]

File: .nvmrc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply .nvmrc:1]: думаю норм

**Review by Yeti-or — APPROVED** [review]

**Review by neretin-trike — APPROVED** [review]

**Review by LamaEats — APPROVED** [review]

**Pull request merged**